poppy/wfe.py

```diff
@@ -299,3 +300,343 @@ def get_opd(self, wave, units='meters'):
+        """ Returns a real-valued random number array of shape (npix, npix)
+        with the symmetry required for a turbulent phase screen.
+        
+        Parameters
+        -----------------
+        npix : int
+            Number of pixels.
+        
+        sign : int
+            Sign of mirror symmetry. Must be either +1 or -1.
+        
+        References
+        -------------------
+        Eq. (65) in J. A. Fleck Jr, J. R. Morris, and M. D. Feit,
+        Appl. Phys. 10, 129 (1976).
+        """
+        
+        sign = float(sign)
+        
+        # create Gaussian, zero-mean, unit variance random numbers
+        a = np.random.normal(size=(npix, npix))
+        
+        # apply required symmetry
+        a[0, int(npix/2)+1:npix] = sign*a[0, 1:int(npix/2)][::-1]
+        a[int(npix/2)+1:npix, 0] = sign*a[1:int(npix/2), 0][::-1]
+        a[int(npix/2)+1:npix, int(npix/2)+1:npix] = sign*np.rot90(a[1:int(npix/2), 1:int(npix/2)], 2)
+        a[int(npix/2)+1:npix, 1:int(npix/2)] = sign*np.rot90(a[1:int(npix/2), int(npix/2)+1:npix], 2)
+        
+        # remove any overall phase resulting from the zero-frequency component
+        a[0, 0] = 0.0
+        
+        return a
+    
+    def rand_turbulent(self, npix):
+        """ Returns a complex-valued random number array of shape (npix, npix)
+        with the symmetry required for a turbulent phase screen.
+        
+        Parameters
+        -----------------
+        npix : int
+            Number of pixels.
+        
+        References
+        -------------------
+        Eq. (63) in J. A. Fleck Jr, J. R. Morris, and M. D. Feit,
+        Appl. Phys. 10, 129 (1976).
+        """
+        
+        # create real-valued random numbers with required symmetry
+        a = self.rand_symmetrized(npix, 1)
+        b = self.rand_symmetrized(npix, -1)
+        
+        # create complex-valued random number with required variance
+        c = (a + 1j*b)/np.sqrt(2.0)
+        
+        return c
+    
+    def power_spectrum(self, wave, L0=None, l0=None, kind='Kolmogorov'):
+        """ Returns the spatial power spectrum.
+        
+        Parameters
+        -----------------
+        wave : wavefront object
+            Wavefront to calculate the power spectrum for.
+        
+        L0 : float
+            The outer scale of the turbulent eddies (m).
+        
+        l0 : float
+            The inner scale of the turbulent eddies (m).
+        
+        kind : string
+            The type of the power spectrum, must be one of 'Kolmogorov',
+            'Tatarski', 'van Karman', 'Hill'.
+        
+        References
+        -------------------
+        G. Gbur, J. Opt. Soc. Am. A 31, 2038 (2014).
+        
+        R. Frehlich, Appl. Opt. 39, 393 (2000).
+        """
+        
+        if not any(kind==item for item in ['Kolmogorov', 'Tatarski', 'van Karman', 'Hill']):
+            raise AttributeError('Kind of power spectrum not correctly defined.')
```

Comment:
`AttributeError` is intended for cases when an object attribute reference fails (e.g. `my_function(some_var.wrong_attribute_name)` -> `object 'some_var' has no attribute 'wrong_attribute_name'`.)

For problems with input parameters like this, more typically the exception to raise would be `ValueError`.